arvo/hoon.hoon

```diff
@@ -10079,7 +10061,16 @@
     ++  wisp                                            ::  core tail
       %-  ulva
       %+  cook
-        |=(a/(list {p/term q/foot}) (~(gas by *(map term foot)) a))
+        |=  a/(list {p/term q/foot})
+        =|  take/(set term)
+        =+  cur=a
+        |-
+          ?~  cur
+            (~(gas by *(map term foot) a))
```

Comment:
The main benefit of |_ is precisely DRY, which imho has
significant functional benefits when not overdone. I suppose there is some
minor performance improvement from keeping the core around for submethods,
but if that becomes relevant you probably want jets.

On Friday, 29 July 2016, Joe Bryan <notifications@github.com> wrote:

> In arvo/hoon.hoon
> <https://github.com/urbit/arvo/pull/199#discussion_r72880835>:
>
> > @@ -10079,7 +10061,16 @@
> >      ++  wisp                                            ::  core tail
> >        %-  ulva
> >        %+  cook
> > -        |=(a/(list {p/term q/foot}) (~(gas by *(map term foot)) a))
> > +        |=  a/(list {p/term q/foot})
> > +        =|  take/(set term)
> > +        =+  cur=a
> > +        |-
> > +          ?~  cur
> > +            (~(gas by *(map term foot) a))
>
> Or (gas:by a b) ;)
>
> Is the :door (|_) an aesthetic thing (a la DRY), or is there some
> functional or performance advantage?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/urbit/arvo/pull/199/files/9a1830065d28917ffe9ca5d0344798c502409ef0#r72880835>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABxXhj5T5482c7aevEuLFKGsbELqHzlKks5qar8egaJpZM4JKm_w>
> .
>